pkg/controller/publishingstrategy/publishingstrategy_controller.go

```diff
@@ -352,6 +359,30 @@ func (r *ReconcilePublishingStrategy) Reconcile(request reconcile.Request) (reco
 	return reconcile.Result{}, nil
 }
 
+// ensure ingresscontrollers on publishingstrategy CR are present on the cluster
+func (r *ReconcilePublishingStrategy) ensureIngressControllersExist(appIngressList []cloudingressv1alpha1.ApplicationIngress, ingressControllerList *operatorv1.IngressControllerList) bool {
+	isContained := true
+	for _, app := range appIngressList {
+		var exists bool
+		for _, ingress := range ingressControllerList.Items {
+			// prevent nil pointer error
+			if ingress.Spec.Domain == "" || ingress.Status.EndpointPublishingStrategy.LoadBalancer == nil {
+				isContained = false
+				break
+			}
+			listening := string(app.Listening)
+			capListening := strings.Title(strings.ToLower(listening))
+			if ingress.Spec.Domain == app.DNSName && capListening == string(ingress.Status.EndpointPublishingStrategy.LoadBalancer.Scope) {
+				exists = true
+			}
+		}
+		if !exists {
+			isContained = false
```

Comment:
Should there be a break here?

I would think that if it's false 1x, we want to return false, not keep going.

Reply:
+1 added break statement